Tests/Packet++Test/Tests/CotpTests.cpp

```diff
@@ -0,0 +1,39 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "CotpLayer.h"
+#include "Packet.h"
+#include "SystemUtils.h"
+
+using namespace std;
+
+PTF_TEST_CASE(CotpLayerTest)
```

Comment:
Let's add tests for `getHeaderLen()` and check that the next layer is a generic payload (`PayloadLayer`)

Reply:
How can I test that the next layer is a generic payload?

Reply:
For `getHeaderLen()` you can do:
```cpp
PTF_ASSERT_EQUAL(newCotpPacket.getHeaderLen(), 3);
```

To check the next layer you can do:
```cpp
PTF_ASSERT_EQUAL(cotpLayer->getNextLayer()->getProtocol(), pcpp::GenericPayload, enum);
```